test/job.js

```diff
@@ -794,21 +790,18 @@ describe('Job', () => {
     it('runs a one-time job after its lock expires', async() => {
       let runCount = 0;
 
-      // @TODO: this lint issue should be looked into: https://eslint.org/docs/rules/no-async-promise-executor
-      // eslint-disable-next-line no-async-promise-executor
-      const processorPromise = new Promise(async resolve =>
-        agenda.define('lock job', {
-          lockLifetime: 50
-        }, async(job, cb) => { // eslint-disable-line no-unused-vars
+      const processorPromise = new Promise(resolve => {
+        agenda.define('lock job', { lockLifetime: 50 }, (job, cb) => { // eslint-disable-line no-unused-vars
           runCount++;
 
           if (runCount !== 1) {
             expect(runCount).to.be(2);
-            await agenda.stop();
-            resolve();
+            agenda.stop().then(() => { // eslint-disable-line promise/prefer-await-to-then
```

Comment:
Another likely not needed `.then()` here.

Reply:
changed to await style